backend/app/assets/javascripts/spree/backend/variant_autocomplete.js

```diff
@@ -21,18 +21,6 @@
       if (typeof(options) === 'object') {
         if (typeof(options['searchParameters']) === 'function') {
           return options['searchParameters'](term)
```

Comment:
As we're not expecting any other type, should we drop the `if` condition?

Reply:
@RyanofWoods I'd love your opinion here, thoughts?

Reply:
Yes, it now makes sense to drop `if (typeof(options['searchParameters']) === 'function')` @kennyadsl, as if they pass a object, `searchParameters` should be mandatory. Please update the JSDocs as well -> `* @param {Function} options.searchParameters`. If they still haven't updated their deprecated code as well, it will be now more useful as it will break rather than silently do default behavior.

Note, this will mean stores cannot explicit pass in a empty object for default behavior, but they should have saw the deprecation for this, and they can just not pass any parameter for this behavior.

Reply:
One more note, this acts a little differently to productPicker, as defaults `options` to `{}` immediately, so we need this `searchParameters` check. Ideally they would have the same structure in the future 🤔 
https://github.com/nebulab/solidus/blob/master/backend/app/assets/javascripts/spree/backend/product_picker.js